src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -377,6 +376,21 @@ class OpCredsFabricTableDelegate : public FabricTableDelegate
                 ChipLogError(Zcl, "OpCredsFabricTableDelegate: Failed to record Leave event");
             }
         }
+
+        // Removing the fabric will cause expiring all associated ReadHandlers so try to send the event ASAP.
+        InteractionModelEngine::GetInstance()->GetReportingEngine().ScheduleUrgentEventDeliverySync();
```

Comment:
So, I'm not certain this event is all that useful. The event doesn't convey which fabric is being deleted, so it's actually not usable by clients.

In addition, scheduling delivery here doesn't guarantee at all that any/all existing subscribers on the to-be-evicted-fabric are going to get this event, for a couple of reasons:
- We may be in the middle of delivering a report to another subscriber and we've hit the limit of number of reports in flight.
- This event may not make it in the to-be generated report since other events have to be first be delivered.

Given how tenous it is that it will actually make it, why even emit this event?

Reply:
Filed https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/5214 to track the ambiguity.

Reply:
Well, I'm OK with removing the event from he spec, but while it's there I thought it would be good to make it work, no? :)

Reply:
Sure. Will approve it for now.

Reply:
Would be good if you could add a comments here about the reasons why it may not be delivered so that at least, others coming across this would understand the tenous nature of this logic.

Reply:
Is there a danger that flushing the event log will consume resources such that the recipient will be less likely to succeed in being able to respond to the RemoveFabric command?

Reply:
@msandstedt The only resource that comes to mind is packet buffer - if we use some for events we may run out of them for the response although that argument holds for any activity (communication on one fabric may consume resources that will make response on another fabric less likely to succeed), so assuming that someone had a good reason putting the leave event to the spec I don't know how to implement it without consuming such resources. I agree it may be another argument for removing the event.

Reply:
Fair enough.  Thanks.

My 2 cents: it is extremely useful for the issuer of the RemoveFabric command to receive a response.  I know there are obstacles to this and that it is particularly difficult when removing the current fabric.  But security policies in many organizations have requirements around removing keys from devices when they are no longer needed.  Receiving the response for RemoveFabric somewhat assists in adhering to such policies.